src/configuration/configuration.ts

```diff
@@ -57,6 +57,28 @@ class ConfigurationClass {
         this[option] = vimOptionValue;
       }
     }
+
+    // Get the cursor type from vscode
+    const cursorStyleString = vscode.workspace.getConfiguration().get("editor.cursorStyle") as string;
+    this.userCursor = this.cursorStyleFromString(cursorStyleString);
+  }
+
+  private cursorStyleFromString(cursorStyle: string): vscode.TextEditorCursorStyle {
+    if (cursorStyle === 'line') {
```

Comment:
I can, I actually copied this code from vscode since I was curious how they did it